docs/guide/vite.md

```diff
@@ -0,0 +1,383 @@
+
+HMR can be configured in three modes using the `@apostrophecms/asset` options:
+
+<AposCodeBlock>
+
+```javascript
+// in app.js
+require('apostrophe')({
+  shortName: 'my-project',
+  modules: {
+    '@apostrophecms/vite': {},
+    '@apostrophecms/asset': {
+      options: {
+        // Choose one of these HMR options:
+        hmr: 'public',  // Default - enables HMR for project UI
+        hmr: 'apos',    // Enables HMR for Admin UI
+        hmr: false      // Disables HMR completely
+      }
+    }
+  }
+});
+```
+ <template v-slot:caption>
+    app.js
+  </template>
+
+</AposCodeBlock>
+
+::: info
+If you use the `hmr: 'apos'` option, you don't need to pass the `APOS_DEV` flag when editing components of the Admin UI. The Vite build will selectively update altered code leading to a better development experience than `APOS_DEV=1`, which will rebuild the entirety of the Admin UI code.
+:::
+
+### WebSocket Configuration
+
+By default, the HMR WebSocket server runs on your ApostropheCMS server port, typically port 3000. For cases where you need a separate port (like running behind certain proxy configurations), you can configure a custom port:
+
+<AposCodeBlock>
+
+```javascript
+'@apostrophecms/asset': {
+  options: {
+    hmrPort: 3001  // Sets a custom WebSocket server port
+  }
+}
+```
+  <template v-slot:caption>
+    app.js
+  </template>
+</AposCodeBlock>
+
+### Source Maps in Production
+
+For debugging production builds, you can enable source maps to see the original source code in your browser's developer tools:
+
+<AposCodeBlock>
+
+```javascript
+'@apostrophecms/asset': {
+  options: {
+    productionSourceMaps: true  // Exposes source maps in production
+  }
+}
+```
+  <template v-slot:caption>
+    app.js
+  </template>
+</AposCodeBlock>
+
+::: tip
+Source maps slightly increase your build size but are invaluable for debugging production issues. Consider your specific needs when enabling this option.
+:::
+
+## Extending Vite Configuration
+
+ApostropheCMS's Vite integration provides two methods to customize your Vite configuration. You can either configure it through your module's code or via a dedicated configuration file in your project root.
+
+### Method 1: Module Configuration
+
+Use the `build.vite` property in your module to extend Vite's configuration:
+<AposCodeBlock>
+
+```javascript
+module.exports = {
+  build: {
+    vite: {
+      myViteConfig: {
+        // Vite configuration options
+        define: {
+          __API_URL__: JSON.stringify('https://api.example.com'),
+          __ANALYTICS_ENDPOINT__: JSON.stringify(process.env.ANALYTICS_URL)
+        },
+        css: {
+          // CSS configuration options
+          preprocessorOptions: {
+            scss: {
+              additionalData: `$injectedColor: orange;`
+            }
+          }
+        },
+        optimizeDeps: {
+          // Dependencies optimization options
+          include: ['lodash', 'moment']
+        },
+        server: {
```

Comment:
Don't recommend that! The https will break the things. The server port will be probably ignored by Vite in middleware mode (the one we use).

Reply:
Removed.